src/types/Stream.d.ts

```diff
@@ -0,0 +1,10 @@
+import Client from './Client'
+import { ExprArg } from './query'
+
+export interface Subscription {
+  on: (type: SubscriptionEvents, callback: Function) => this
+  start: () => this
+  close: () => this
+}
+
+type SubscriptionEvents = 'ref' | 'ts' | 'new' | 'old' | 'diff' | 'action'
```

Comment:
These are fields that come with the "version" event. The event types are: "start", "version", "history_rewrite", and "error". The list of events can be found in the docs here: https://github.com/fauna/faunadb-js/blob/master/src/stream.js#L289-L372

Here is the full list of events and their fields for convenience:

```
event: start
  fields:
    - event: the event name
    - txnTS: the event transaction timestamp
    - data: the stream start timestamp

event: version
  fields:
    - event: the event name
    - txnTS: the event transaction timestamp
    - data:
      - ref: the document's ref
      - ts: the document's valid timestamp
      - action: the action that triggered the event
      - new: the document's new data
      - old: the document's old data (optional)
      - diff: the diff between old and new data (optional)

event: history_rewrite
  fields:
    - event: the event name
    - txnTS: the event transaction timestamp
    - data:
      - ref: the document's ref
      - ts: the document's valid timestamp
      - action: the action that triggered the event

event: error
  fields:
    - event: the event name
    - txnTS: the event transaction timestamp
    - data:
      - code: the error code
      - description: the error description
```